### DISCUSSION TITLE
Target Values (Y) in ART Generate Function

### DISCUSSION THREAD

**Discussion by fatimah-aloraini** [Trusted-AI/adversarial-robustness-toolbox#2001] [Q&A] [answered]
Dear developers, 
Thank you for making ART available to everyone, it is such a great help. 

When crafting adversarial examples using gradient-based methods such as FGM, JSMA, BIM, etc. One of the arguments in generate function is **Y** which is as described by **ART** : _Target values (class labels) one-hot-encoded of shape (nb_samples, nb_classes) or indices of shape (nb_samples,)Only provide this parameter if you’d like to use true labels when crafting adversarial samples. Otherwise, model predictions are used as labels._ 

**My question is** when providing **Y** to generate function should we provide the true labels (original labels) or the target label ? 

I really appreciate any help you can provide.
Warm regards, 
Fatimah

**Comment by beat-buesser** [comment, marked as answer]
Hi @fatimah-aloraini Thank you very much for using ART!

The question if true labels or target labels should be provided depends on whether the attack is being run as un-targeted attack (attack argument `targeted=False`, default for most attacks) or as targeted attack (attack argument `targeted=True`). Un-targeted attacks expect true labels whereas targeted attacks expect target labels.

**Reply by fatimah-aloraini** [reply]
Thank you for your clarification,
appreciate your help,
Fatimah
